File: 03-frontend/angular-ecommerce/src/app/common/cart-item.ts
import { Product } from "./product";

export class CartItem {

    id!: string;
    name: string;
    imageUrl: string;
    unitPrice: number;
    quantity: number;
    productId!: string;
    customerId!: string;

    constructor(product: Product){
        this.productId = product.id;
        this.name = product.name;
        this.imageUrl = product.imageUrl;
        this.unitPrice = product. unitPrice;
        this.quantity = 1;
    }


}

File: 02-backend/spring-boot-ecommerce/src/main/java/com/ecommerce/entity/CartItem.java
package com.ecommerce.entity;

import java.math.BigDecimal;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "cart_item")
@Getter
@Setter
public class CartItem {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "id")
	private Long id;
	
	@Column(name = "name")
	private String name;
	
	@Column(name = "image_url")
	private String imageUrl;
	
	@Column(name = "unit_price")
	private BigDecimal unitPrice;
	
	@Column(name = "quantity")
	private int quantity;
	
	@Column(name = "product_id")
	private Long productId;
	
	@Column(name = "customer_id")
	private Long customerId;
	
	
}
